80 reporting

Juan-Mateos opened the pull request:
Closes #80

Commit:
Added introduction chapter

Commit:
Added introduction chapter

Commit:
Added introduction chapter

Commit:
Added literature section and bib

Commit:
Added literature section and bib

Commit:
Added literature section and bib

Commit:
Changed names of markdown files

Commit:
Changed names of markdown files

Commit:
Merge branch '80_reporting' of github.com:nestauk/sg_covid_impact into 80_reporting

Commit:
Changed markdown names

Commit:
Changed markdown names

Commit:
New section

Commit:
New section

Commit:
New references

Commit:
New references

Commit:
Work in methodology

Commit:
New references

Commit:
machine learning and triangulation figures

Commit:
Added glass map

Commit:
Finished methodology section

bishax committed:
Make report cmd, rename bib,title change,figure paths,

Commit:
code for new figure and updated figures

Commit:
Changed model_config of lads to show in map and figure output

Commit:
Changed colour palette in one of the charts, add glass validation script

Commit:
reports md

bishax committed:
HTML report with interactive figures setup.

bishax committed:
Fix altair pandoc for pdf output.

bishax committed:
new altair md syntax passes style info + example

Commit:
New functions for reporting

Commit:
Merge branch '80_reporting' of github.com:nestauk/sg_covid_impact into 80_reporting

bishax committed:
Sketch out fuzzy matching section; placeholders for notice content.

Commit:
results section and scripts

Commit:
Add descriptive script

Commit:
section 0 and section 5

bishax commented on reports/technical_report/1_introduction.md:
FRON biases

bishax commented on reports/technical_report/2_literature.md:
```diff
@@ -0,0 +1,57 @@
+
+## Business panels
+
+Business panels are regular voluntary surveys including small representative samples of firms. An example is the [Business Impact of Coronavirus Survey (BICS)](https://www.ons.gov.uk/economy/economicoutputandproductivity/output/datasets/businessinsightsandimpactontheukeconomy), which is undertaken on a fortnightly basis and includes questions about company financial status, its resort to governme ...
```
FORLOGH Scheme

bishax replied:
recent RELASE of

bishax commented on reports/technical_report/2_literature.md:
```diff
@@ -0,0 +1,57 @@
+In December 2020, Scottish Government published a [comprehensive report about Covid-19's impact in Scotland](https://nationalperformance.gov.scot/sites/default/files/documents/NPF_Impact_of_COVID-19_December_2020.pdf) including a section about the impact on "the economy, fair work and business culture" which draws on similar sources to those we described above:
+
+* BICS data from November shows that "Arts, Entertainment and Recreation" and "Accommodation and Food Services" have been most impacted by the pandemic, with 11% of the former and 38% of the latter temporarily closed. These sectors also have experienced the greatest decline in turnover and have the largest share of employees furloughed.
+* The report features [an analysis by the Fraser Allander Institute](https://fraserofallander.org/mapping-local-job-vulnerability/) that combines information about the impact of Covid-19 on sectoral GDP in Scotland and information about the geography of sectoral employment to measure the vulnerability to Covid-19 in different parts of Scotlan [@mcintyre2020vulnerability]. One important finding of ...
```
different parts of SCOTLAN

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+The starting point for our analysis is that business websites contain information that may be relevant for the analysis of the economic impact of Covid-19. This includes descriptions of what businesses do which may help us to quantify their exposure / opportunities to diversify from Covid-19 and Covid-19 notices where businesses describe how they are adapting in response to the pandemic. It is po ...
+
+At the same time, business websites are hard to use for longitudinal analyses because they do are not updated following a regular schedule, they lack policy-relevant information about employment and turnover, and suffer from potential biases in their sectoral and geographical coverage.
```
they DO are not [extra word]

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+1. We can use Google Search trend data queried with the names of products and services that businesses sell (extracted from their business descriptions) to build a longitudinal picture of exposure to Covid-19. We can also extract social media profiles from business websites and use them as a seed to collect their social media feeds in order to generate more regularly updated measures of their act ...
+2. We can combine measures of sectoral exposure to Covid-19 based on the text from business websites with official data to estimate the share of workorce in local economies that is exposed to Covid-19. This is similar to the approach in [@del2020supply, @mcintyre2020vulnerability, @enenkel2020cities] but with the advantage of relying on data-driven measures of sectoral exposure to Covid-19 derive ...
+3. We can use the text in business websites to generate maps of "sectoral proximity" helping us to identify opportunities for industrial diversification away from markets that are highly exposed to Covid-19. We can use the results to rank the position of different sectors in terms of their ability to diversify away from Covid-19, and combine this with official data to measure the extent to which  ...
+4. We can combine business website information with the business registry in order to track business outcomes (such as business failure) ane explore opportunities to "nowcast" them using their text and Covid-19 notices as predictorrs (the idea being that there may be some signal about a business' outcomes in the text that describes current behaviours in response to the pandemic). 
```
ANE explore

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+|80: Security and investigation activities | police security cctv sia investigation guards alarms locksmith surveillance security\_systems access_control fire\_security security\_guards |
+|25: Manufacture of fabricated metal products, except machinery and equipment | metal steel engineering wire gates stainless\_steel aluminium components assembly welding precision cnc aerospace fabrication sheet\_metal | 
+
+Table: Industrial vocabulary examples by SIC division \label{tab:examples}
```
The pandoc syntax to get the referencing and caption right for any format would be (note the colon at the beginning and the fact that the label prefix needs to be `tbl:`):
`: Industrial vocabulary examples by SIC division {#tbl:examples}`
This should be placed on a line directly below the last row of the table itself.

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+In order to query Google Trends, we need a list of keywords related to the economic activities of different industries. We extract this "industry vocabulary" from Glass business descriptions. Our strategy is to aggregate the descriptions of all businesses in the same SIC division, tokenise then (extract individual works and commonly occurring combinations of words), count them and normalise these ...
+
+Table \ref{tab:examples} presents examples of industrial vocabularies randomly extracted from our list of SIC divisions. It shows that in general the approach we have taken appears to work well and generates intuitive vocabularies for various divisions, although in some cases such as with division 21 (pharmaceuticals), it generates very few keywords, potentially reducing the robustness of our sea ...
```
`[@tbl:examples]`

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+We extract normalised search volumes for keywords in industry vocabularies from Google Search Trends and use the results to produce measures of industrial exposure to Covid-19. In order to do this we:
+
+1. Take weekly search volumes by keyword to generate monthly averages.
+2. Normalise the search volume of each keyword by its salience in a SIC division. This means that keywords that are more salient for the industry will have a stronger weight when estimating its exposure to Covid-19 than terms that are less salient (ie occur more frequently in other divisions). For example, in Division 25 in table \ref{tab:examples} `metal` will be more salient (and have more weig ...
```
`[@tbl:examples]`

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+1. Take weekly search volumes by keyword to generate monthly averages.
+2. Normalise the search volume of each keyword by its salience in a SIC division. This means that keywords that are more salient for the industry will have a stronger weight when estimating its exposure to Covid-19 than terms that are less salient (ie occur more frequently in other divisions). For example, in Division 25 in table \ref{tab:examples} `metal` will be more salient (and have more weig ...
+3. Normalise search volumes by the total volume of searches in the industry vocabulary. This means that keyords that account for a larger proportion of all searches in an industry vocabulary will have a bigger weight in determining its exposure to Covid-19.
```
that KEYORDS that

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+1. Take weekly search volumes by keyword to generate monthly averages.
+2. Normalise the search volume of each keyword by its salience in a SIC division. This means that keywords that are more salient for the industry will have a stronger weight when estimating its exposure to Covid-19 than terms that are less salient (ie occur more frequently in other divisions). For example, in Division 25 in table \ref{tab:examples} `metal` will be more salient (and have more weig ...
+3. Normalise search volumes by the total volume of searches in the industry vocabulary. This means that keyords that account for a larger proportion of all searches in an industry vocabulary will have a bigger weight in determining its exposure to Covid-19.
+4. Rescale 2020 and 2021 normalised keyword search volumes by the normalised search volumes in equivalent 2019 months in order capture deviations from a pre-Covid-19 baseline and account for seasonality in searches (ie the fact that people are more likely to search for information about air transport when they are planning their holidays at the beginning of the year, or for information about spor ...
```
in order TO [missing word]

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+### d. Calculating sectoral diversification options
+
+We use similarity between business website descriptions in order to calculate similarities between industries, which we interpret as a proxy for diversification opportunities: those sectors that are "close" to each other within the industry space that we build are more likely to be able to diversify into each other activities (this idea is based on the "Principle of Relatedness" which economic ge ...
```
into each other'S activities [missing possessive]

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+We use similarity between business website descriptions in order to calculate similarities between industries, which we interpret as a proxy for diversification opportunities: those sectors that are "close" to each other within the industry space that we build are more likely to be able to diversify into each other activities (this idea is based on the "Principle of Relatedness" which economic ge ...
+
+In order to calculate similarities between industries we train a machine learning model that predicts the SIC divisions for companies in the Glass-Companies House matched dataset using their descriptions after standard pre-proxessing. We use grid-search to select the model type and parametres that yield the best predictive performance, which in this case is a regularised logistic regresssion. 
```
pre-proXessing

bishax replied:
PARAMETRES

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+![Figure Predictive model](../../figures/scotland/png/appendix_model_validation.png){#fig:model}
+
+[@fig:model] summarises the predictive performance of the model for different SIC divisions. The model includes the actual label in the top five predictions it makes 72% of the times, and in the top ten 84%. This feels reasonable given the degree of noise in company descriptions and low level of informativeness in some SIC labels such as "Other activities not elsewhere classified" but there is st ...
```
Also the possibility for companies house SIC labels to be incorrect.

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+Since our model has been trained using a one vs rest classification regime that generates a probability score for each business and SIC division, we can use the resulting vector to calculate "industry co-occurrences" in a company (ie companies that are predicted to belong to more than one sector based on the machine learning model). 
+
+We create a network (the "industry space") where the nodes are SIC divisions and the edges are co-occurrences in business descriptions (the degree or thickness of these edges depends on the number of co-occurrences) and arrange it using a force-directed algorithm that bundles more closely those nodes which are similar to each other. In order to simplify our visualisations, we will use a maximum s ...
```
May be worth a brief comment on the difference between our industry space and Hidalgo's product space, i.e. our co-occurrence is at the firm-level, whereas theirs was based on co-occurrence within geographical regions.

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+![Comparison of Glass and Companies House sectoral coverage](../../figures/scotland/png/glass_sector_validation.png){#fig:sector_val}
+
+The figure shows that *Primary sectors* and *Transportation* tend to be underrepresented in the Glass data while *Manufacturing*, *Administrative services*, *Health*, the *Arts* and *Professional Activities* tend to be overrepresented. *Activities of Membership Organisations* is an outlier in terms of representation probably because of the presence of some charity data in Glass.
```
Required extra information: Activities of Membership Organsiations is in Section S

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+The figure shows that *Primary sectors* and *Transportation* tend to be underrepresented in the Glass data while *Manufacturing*, *Administrative services*, *Health*, the *Arts* and *Professional Activities* tend to be overrepresented. *Activities of Membership Organisations* is an outlier in terms of representation probably because of the presence of some charity data in Glass.
+
+In [@place_val] we perform a similar analysis but comparing the geographical representation of different Local Authority Districts in Glass and Companies House. 
```
`[@fig:place_val]`

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+![Comparison of Glass and Companies House geographical coverage](../../figures/scotland/png/glass_place_validation.png){#fig:place_val}
+
+We see that Local Authorities in Highlands and Islands tend, in general, to be overrepresented in the Glass data while Local Authorities around Aberdeen and the central belt - in particular Glasgow City - are underrepresented. We dig further into the potential reasons for this in [@sector_place_val], where we present, in the left panel, the over or underrepresentations of SIC divisions (horizonta ...
```
`[@fig:sector_place_val]`

bishax replied:
Local authorities in Scotland should be referred to as "Council areas" I believe.

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+![Evolution in keyword search trends](../../figures/scotland/png/keyword_trends.png){#fig:keyword_trends}
+
+The chart, which can be explosed [here](add link) shows that at the beginning of the pandemic terms such as *symptoms* and *news* attracted unusual (versus the pre-Covid-19 baseline) levels of attention. By contrast, terms related to travelling, sports and social consumption such as *airport*, *bus*, *football* or restaurant experience a drop compared to the same month in the previous year. Certa ...
```
can be EXPLOSED

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+[@fig:sector_exposures] presents monthly changes in sectoral exposure to Covid-19. Every row in the Y axis is a SIC division (not all of them are labelled - the chart can be explored interactively [here](add link)). The colour of each cell represents the position of a SIC division in the ranking of exposure to Covid-19. Red colours indicate high levels of exposure (ie low volumes of search for ke ...
+
+![Evolution in sectoral exposure](../../figures/scotland/png/sector_exposures.png){#fig:sector_exposures}
```
Is there enough width left on the page to give even truncated descriptions of the divisions? Otherwise it's hard to interpret.

bishax replied:
If not then the paragraph below should perhaps include division numbers too

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+As an additional validation step, we have correlated time-series of search interest on products and services related to different industries aggregated at the SIC section level with other trend data indicative of the severity of Covid-19 and its impact on individual behaviours proxied via Google Footfall data. We present the results in [@fig:add_val], where every facet presents the correlation co ...
+
+![](../../figures/scotland/png/trend_triangulation.png){#fig:add_val}
```
Figures need some caption text in order to be built as proper figures.

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+![](../../figures/scotland/png/trend_triangulation.png){#fig:add_val}
+
+The chart shows that volumes of interest in sector requiring high levels of physical proximity and social interaction such as *Construction*, *Arts and Entertainment*, *Transport* or *Real Estate* are negatively correlated with proxies for the impact of the severity of the pandemic (and allied social distancing and lock down measures). They are also negatively correlated with measures of consumer ...
```
in sectorS

bishax commented on reports/technical_report/3_methodology.md:
```diff
@@ -0,0 +1,196 @@
+
+The chart shows that volumes of interest in sector requiring high levels of physical proximity and social interaction such as *Construction*, *Arts and Entertainment*, *Transport* or *Real Estate* are negatively correlated with proxies for the impact of the severity of the pandemic (and allied social distancing and lock down measures). They are also negatively correlated with measures of consumer ...
+
+The sign of these correlation are reversed for *Information and Communication* activities aand *Manufacturing* where consumption is less dependent on physical proximity. 
```
Aand

bishax commented on bin/make_report.sh:
```diff
@@ -0,0 +1,15 @@
+set -euo pipefail
+
+cd $(dirname $0)/../reports/technical_report
+pandoc -s 1_introduction.md 2_literature.md 3_methodology.md\
```
Need to add 4_results.md and conclusions.md to the source listings

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+[@fig:nat_evol] presents the evolution and composition of employment in sectors highly exposed to Covid-19 (decile 8 and above of exposure) between March 2020 and January 2021.
+
+* The top panel presents aggregate shares of employment in Scotland and in the rest of the UK. It shows similar patterns of economic exposure to Covid-19 in terms of levels and evolution. We see spikes of exposure around the three main Covid-19 waves and associated lockdowns / policy restriction (March, October and Christmas 2020).
+* The bottom panel presents the shares of employment in Scotland exposed to Covid-19 accounted for by different SIC sections (the [interactive version](link here) allows a more finely grained exploration at the SIC divisionslevel). The charts shows important changes in the composition of workforces exposed to the pandemic at different points: during the first lockdown a significant share of the e ...
```
phases of thE pandemic

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+  * In Manufacturing, Scotland's two main areas of competitive advantage (*Manufacture of beverages* and *Electricity*) do not appear to be substantially exposed to Covid-19
+  * In Services, Scotland is specialised in three sectors with high levels of exposure to Covid-19 - they are *Libraries and museums*, *Accommodation* and *Architectural and Engineering Activities*. 
+
+![Evolution and composition of exposure to Covid-19](../../figures/scotland/png/exposure_shares_Scotland.png){#fig:nat_comp}
```
Could do with division numbers / names

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+We see that:
+
+* In April, the Highland council areas was strongly exposed to Covid-19 because of its reliance on *Accommodation**, *Wholesale and Retail trade of Automobiles* and *Specialised Construction Activities*.
```
*Accomodation** [extra asterisk]

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+#### Local exposure
+
+We display the evolution in the share of the workforce in sectors highly exposed to the pandemic in [@fig:ca_evol]. Each line corresponds to a council area and the colour of the lines represents the mean exposure share over the whole period [explore individual trends in the interactive](here). 
```
The `here` needs to be in the text too.

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+We display the evolution in the share of the workforce in sectors highly exposed to the pandemic in [@fig:ca_evol]. Each line corresponds to a council area and the colour of the lines represents the mean exposure share over the whole period [explore individual trends in the interactive](here). 
+
+We see that the evolution of exposure to Covid-19 in council areas reflects the national trends in [@fig:nat_evol]: a spike in economic exposure from April slowly declining until August and restarting in September . In terms of individual council areas,  Council Areas in the highlands such as Argyll and Butte Perth and Kinross and Stirling, as well as some of the islands appear more highly expose ...
```
Council area names need some commas between them

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+We see that:
+
+* In April, the Highland council areas was strongly exposed to Covid-19 because of its reliance on *Accommodation**, *Wholesale and Retail trade of Automobiles* and *Specialised Construction Activities*.
+* In September, Aberdeen appears strongly exposed to Covid-19 because of the importance of *Extraction of Crude, Petroleum and Gas*, *Exployment activities* and *Legal and Accounting activities* among other sectors.
```
eXployment activities

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+The figure illustrates some of the diversification opportunities / challenges for sectors that are highly exposed to Covid-19:
+
+![Scottish sector space](../../figures/scotland/png/sector_space_Scotland.png){#fig:sector_space}.
```
The period at the end of this line, prevents the figure being picked up properly.

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+* *Creative arts and entertainment* (towards the top) can diversify into *Motion picture activities*.
+* *Accommodation* activities is close to *Real Estate activities* - unfortunately this is a sector that appears as highly exposed to Covid-19 in January 2021.
+
+In order to summarise a sector's diversification options away from Covid-19, we measure how far away it is, on average, from sectors with low levels of exposure to Covid-19. We present the results in Chart [@fig:div_option)[interactive version](link here). In that chart, the vertical axis represents SIC divisions sorted by their average distance to less exposed industries in January 2021, with in ...
```
`[@fig:div_option)` - mismatched brackets

space needed between reference and subsequent hyperlink.

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+In order to summarise a sector's diversification options away from Covid-19, we measure how far away it is, on average, from sectors with low levels of exposure to Covid-19. We present the results in Chart [@fig:div_option)[interactive version](link here). In that chart, the vertical axis represents SIC divisions sorted by their average distance to less exposed industries in January 2021, with in ...
+
+![Scottish sector space](../../figures/scotland/png/sector_diversification_options.png){#fig:div_options}.
```
Needs division numbers/names

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+#### Evolution of diversification
+
+In Figure [@fig:divnat] we show the evolution of shares of employment in sectors that are highly exposed to Covid-19 but have low diversification options (top panel) comparing Scotland with outside, and looking into the specific industries that are contributing to Scotland's high exposure (bottom panel). 
```
`Figure` not needed

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+![Evolution of diversification](../../figures/scotland/png/div_trends.png){#fig:divnat}
+
+In [@fig:localdiv] we map shares of employment in "high exposure - low diversification" sectors in council areas for selected months and council areas. The maps in the top panel suggest a degree of persistence in the council areas that appear more exposed to Covid-19 and less able to diversity from it - these are particularly in the Highlands. In the bottom panel we display the "industry spaces"  ...
```
"these are particularly in the Highlands" [grammar/missing words]

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+Our secondary variables include measures of the economic health of a location (employment rate and median annual gross income), level of education (including share of population with no qualifications and share with tertiary education) and a measure of economic complexity that captures the sophistication of the productive capabilities in a location [@bishop2019exploring].
+
+Before exploring the link between all these variables, we present the evolution of normalised claimant count rates by council area in Scotland in [@fig:cl_scot] ([explore individual council area trends in the interactive version](link here)). The evolution of the series is consistent with the idea of a Covid-19 impact after March 2020 with an increase in claimant count rates *compared with the sa ...
```
Something is wrong with the rendered citation of `resfund2020truth` - it appears as "[res, 2020]"

bishax replied:
There was an error in the `.bib` - see file

bishax commented on reports/technical_report/technicalreport.bib:
```diff
@@ -0,0 +1,144 @@
+@report{resfund2020truth,
+title={The truth will out: Understanding labour market statistics
+author={Brewer, Mike and Gardiner, Laure and Handscomb, Karl}
+during the coronavirus crisis},
```
This lone belongs above author, as part of the title.

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+Before exploring the link between all these variables, we present the evolution of normalised claimant count rates by council area in Scotland in [@fig:cl_scot] ([explore individual council area trends in the interactive version](link here)). The evolution of the series is consistent with the idea of a Covid-19 impact after March 2020 with an increase in claimant count rates *compared with the sa ...
+
+When we check the evolution of individual council areas, we find that affluent council areas such as City of Edinburgh and Aberdeen City have experienced a stronger shock from Covid-19, with claimant count rates three times higher than the same month in the previous year). This supports the notion already highlighted during the literature review that the pandemic has had a stronger impact on urba ...
```
WHITSTAND

bishax replied:
It only takes a small addition of claimants in wealthier areas (with lower baseline claimant count levels) to see large multiplicative factors.

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+![Scottish sector space](../../figures/scotland/png/claimant_counts_Scotland.png){#fig:cl_scot}
+
+This observation is further confirmed by the correlation matrix in [fig:corr_tab], where we present correlations between relevant variables in our data with a focus on Scottish council areas (warmer colour represent positive correlations). We find that:
```
`[@fig:corr_tab]`

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+![Scottish sector space](../../figures/scotland/png/correlation_table.png){#fig:corr_tab}
+
+We explore further these patterns in [@fig:biv], where we visualise the bivariate association between our measures of exposure and diversification opportunities (focusing on current exposure instead of exposure in previous months) and secondary variables (we have . The colour of the points represent the normalised claimant count for a council area. We also display the slope of the regression coef ...
```
"( we have ." [truncated sentence?]

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+We explore further these patterns in [@fig:biv], where we visualise the bivariate association between our measures of exposure and diversification opportunities (focusing on current exposure instead of exposure in previous months) and secondary variables (we have . The colour of the points represent the normalised claimant count for a council area. We also display the slope of the regression coef ...
+
+Although the relationships between variables are generally weak, we note with interest a positive link between our exposure measure to Covid-19 and the level of education (share of population with tertiary education). The same link is negative when we focus on exposure meaasures that consider only employment in sectors with less diversification options. One interpretation of this result is that c ...
```
"Although the relationships between variables are generally weak, we note with interest a positive link between our exposure measure to Covid-19 and the level of education (share of population with tertiary education)."

0.02? Really?

bishax replied:
exposure MEAASURES

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+![Correlations between exposure / diversification shares and secondary data](../../figures/scotland/png/bivariate_scatters.png){#fig:biv}
+
+The analysis above has highlighted the complexity of the interrelationships between our variables: wealthy and more educated council areas have experienced bigger increases in claimant counts compared to a pre-pandemic baseline. They appear to be somewhat more exposed to Covid-19 but in sectors that may be able to diversify into other industries less exposed to the pandemic. We incorporate all th ...
```
normalised claiNmant

bishax replied:
economic complexTIy

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+The analysis above has highlighted the complexity of the interrelationships between our variables: wealthy and more educated council areas have experienced bigger increases in claimant counts compared to a pre-pandemic baseline. They appear to be somewhat more exposed to Covid-19 but in sectors that may be able to diversify into other industries less exposed to the pandemic. We incorporate all th ...
+
+![Scottish sector space](../../figures/scotland/png/modelling_results.png){#fig:regression}
```
Labels converted from variable name to description name?

bishax commented on reports/technical_report/4_results.md:
```diff
@@ -0,0 +1,130 @@
+
+![Correlations between exposure / diversification shares and secondary data](../../figures/scotland/png/bivariate_scatters.png){#fig:biv}
+
+The analysis above has highlighted the complexity of the interrelationships between our variables: wealthy and more educated council areas have experienced bigger increases in claimant counts compared to a pre-pandemic baseline. They appear to be somewhat more exposed to Covid-19 but in sectors that may be able to diversify into other industries less exposed to the pandemic. We incorporate all th ...
```
What is the lag?

bishax commented on reports/technical_report/5_conclusions.md:
```diff
@@ -0,0 +1,62 @@
+
+Our analysis of exposure and diversification to Covid-19 in Scotland suggests that the industrial composition of the country makes it slightly more succeptible to the economic impacts of Covid-19 than other parts of the UK: sectors related to tourism such as *Accommodation* and *Travel agencies* and cultural activities such as *Libraries and Museums* where Scotland has a relative specialisation a ...
+
+This is reflected in important geograaphical heterogeneity in exposure to Covid-19. While council areas in the Highlands and Islands appear to be more exposed to Covid-19 because of the relative importance for their economies of industries such as those we described above, we also see council areas in other parts of Scotland showing high levels of exposure to the pandemic as a consequence of thei ...
```
GEOGRAAPHICAL

bishax commented on reports/technical_report/5_conclusions.md:
```diff
@@ -0,0 +1,62 @@
+
+Our analysis of exposure and diversification to Covid-19 in Scotland suggests that the industrial composition of the country makes it slightly more succeptible to the economic impacts of Covid-19 than other parts of the UK: sectors related to tourism such as *Accommodation* and *Travel agencies* and cultural activities such as *Libraries and Museums* where Scotland has a relative specialisation a ...
+
+This is reflected in important geograaphical heterogeneity in exposure to Covid-19. While council areas in the Highlands and Islands appear to be more exposed to Covid-19 because of the relative importance for their economies of industries such as those we described above, we also see council areas in other parts of Scotland showing high levels of exposure to the pandemic as a consequence of thei ...
```
Local authorities -> Council areas

bishax replied:
which more manufacturing activity appear -> where more manufacturing activity appears?

bishax commented on reports/technical_report/5_conclusions.md:
```diff
@@ -0,0 +1,62 @@
+
+This is reflected in important geograaphical heterogeneity in exposure to Covid-19. While council areas in the Highlands and Islands appear to be more exposed to Covid-19 because of the relative importance for their economies of industries such as those we described above, we also see council areas in other parts of Scotland showing high levels of exposure to the pandemic as a consequence of thei ...
+
+Our analysis of the evolution of normalised claimant counts at the council area level and their correlation with other secondary data suggests that wealthier local economies with better educated workforces have suffered a stronger shock from Covid-19. At the same time, these areas are likely to have access to the resources and capabilities to overcome this shock, not least the skills to diversify ...
```
It is most likely the poorer people in the wealthy economies who suffer the shock.

bishax commented on reports/technical_report/5_conclusions.md:
```diff
@@ -0,0 +1,62 @@
+
+Our industry vocabularies are based on the business website descriptions of all companies in the UK, which might miss products and services that Scotland specialises on within particular industries. We could address this by re-building the business vocabularies using only the descriptions from Scottish companies. One potential issue with this approach is that it would reduce the sample sizes we u ...
+
+We have extracted Google Search trend data for all the UK under the assumption that these trends will be broadly aligned with the evolution of the situation in Scotland. We could however re-run our queries to extract search trends at the level of Scotland or even sub-national geographies. This might be particularly relevant for the analysis of exposure to Covid-19 in sectors with a strong level o ...
```
Note: requires methodological R&D to be able to put different geographies on the same absolute scale (GTAB doesn't do this)

bishax commented on reports/technical_report/5_conclusions.md:
```diff
@@ -0,0 +1,62 @@
+
+We have extracted Google Search trend data for all the UK under the assumption that these trends will be broadly aligned with the evolution of the situation in Scotland. We could however re-run our queries to extract search trends at the level of Scotland or even sub-national geographies. This might be particularly relevant for the analysis of exposure to Covid-19 in sectors with a strong level o ...
+
+On the modelling side, we have shown that present and lagged measures of exposure to Covid-19 and opportunities to diversify away from Covid-19 are linked to claimant count rate and normalised claimant count rate levels. One could explore different modelling strategies and additional data sources to attempt to nowcast claimant count rates. One challenge for thi effort is that - as we have shown t ...
```
for THI effort

bishax replied:
could CREAATE

bishax commented on reports/technical_report/5_conclusions.md:
```diff
@@ -0,0 +1,62 @@
+
+### Visualisation options
+
+In this project we have worked with datasets with a high degree of temporal, geographical and sectoral granularity. Although we have tried to showcase this richness through the interactive charts linked throughout the report, we believe that realising their potential requires the development of dedicated dashboards and interactive tools. These would display regularly updated information about the ...
```
POLICYMNAKERS

bishax reviewed (commented):
Throughout `ie` -> `i.e.`

A vague thought: "Descriptive analysis and validation" in section 3 feels more at home in section 4?

Commit:
Prediction chart

Commit:
Addressing PR comments

Commit:
Addressing PR comments

Commit:
Added revised figures

Commit:
Changes in conclusion

Juan-Mateos commented:
I have addressed all your comments.

Some observations:
* All figures with divisions in Y axis have numbers now. I haven't included their names because that would be quite hard to read. To address this we can: (1) include a lookup table between division numbers and names at the end of the report and include the numbers to mention of sectors in the text; (2) point users to the html versions (3): mark up the pngs to highlight interesting results. 2 is already done and 1 would be easy. I think we should also do 3 but not until we are about to publish the report so as to avoid having to redo laborious hand-editing of charts if these change.
* I removed the bivariate comparisons which as you said were weak.
* I note your point about the location of the descriptive results in the validation section. The reason why I put it in the methodology section was to make sure that the findings focused on policy-relevant results for Scotland. 

On Monday I will write the executive summary and add links throughout.

bishax committed:
Merge notice analysis and twitter analysis frm dev

bishax committed:
Blacken.

bishax committed:
Notice,twitter,business data; topic modelling method.

bishax committed:
Notice analysis report section.

bishax committed:
Twitter analysis report section.

bishax committed:
Micro conclusion draft.

bishax committed:
Update html report make script.

bishax committed:
Fix missing references.

bishax committed:
Fix more references.

bishax committed:
Notice and twitter edits; random reference fixes.

bishax committed:
Fix pdf report generation.

bishax committed:
Make HTML report self-contained for easy inclusion of PNG's in s3 report version

Juan-Mateos commented on reports/technical_report/3_methodology.md:
```diff
@@ -42,9 +50,26 @@ We collect search volume data for a set of keywords extracted from business webs
+
+- Discard any twitter accounts mentioned more than 5 times across all websites
+- If there is only one match, consider that a match (high recall, low precision).
+- If one of the candidate twitter accounts has a string similarity score of over 70% then choose as a match.
```
say: "has a string similarity score to the company name of over 70%"

Juan-Mateos commented on reports/technical_report/3_methodology.md:
```diff
@@ -69,25 +94,38 @@ We have collected additional data from a number of secondary sources which we us
+Granularity is important because it determines the level at which decisions can be made.
+Timeliness is important because it allows us to detect drift in the validity of indicators (as the pandemic and the response to it evolve) and increases the data available to validate and train models against.
+
+Whilst trying to identify sources of business failure data we faced two primary problems.
```
Give examples of the ones we scoped: "We identified the following potential data sources of business failure: X, Y, Z" and briefly (one line?) describe what they are and add a link. Then: "Our assessment of these data sources revealed two primary problems..."

Juan-Mateos commented on reports/technical_report/3_methodology.md:
```diff
@@ -69,25 +94,38 @@ We have collected additional data from a number of secondary sources which we us
+
+The matching methodology matches the names of companies in Companies House with the names extracted by Glass from business websites.
+Naively comparing the similarity of names is computationally infeasible.
+ We therefore use Locality Sensitive Hashing (with the Jaccard similarity of k-shingles) and the cossine similarity of tokens to identify the most likely matches for each Glass website which we then apply an exact similarity measure to and choose the best match.
```
typo in cossine (cosine).

Would it be possible to briefly explain locality sensitive hashing to a policy audience even if crudely ("uses a method from database design to identify company names that are similar to each other") ?

Juan-Mateos commented on reports/technical_report/3_methodology.md:
```diff
@@ -138,7 +176,35 @@ In order to estimate a sector's diversification options to reduce exposure to Co
-[content to be inserted from notice_analysis.md in branch 17_notice_topic_model]
+#### Pre-processing
+
+To turn the notice text into data we can mathematically analyse, we process the notices into tri-grams performing filtering and lemmatising as we go.
```
explain what are tri-grams and lemmatising

Juan-Mateos commented on reports/technical_report/3_methodology.md:
```diff
@@ -138,7 +176,35 @@ In order to estimate a sector's diversification options to reduce exposure to Co
+We train a TopSBM [@topSBM] topic model on our pre-processed collection of Covid notices
+ from Scottish business websites.
+This approach confers multiple advantages
+ over the more traditional LDA [@LDA]
```
"Latent Dirichlet Allocation (LDA) frequently used in the literature"

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -139,3 +139,125 @@ We note that the estimates above are experimental and developed for illustrative
+
+Figure [@fig:notice_proportion_section] paints a more interesting picture.
+Construction business websites were much less likely to post notices than would be expected.
+Whilst this could be because construction companies were less likely to have been impacted by Covid from May, the more likely explanation is that construction company websites are not where clients of these companies go to get information - updates are likely to come through more channels.
```
To be more specific, replace "more" with "other channels such as word of mouth"?

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -139,3 +139,125 @@ We note that the estimates above are experimental and developed for illustrative
+Figure [@fig:notice_proportion_section] paints a more interesting picture.
+Construction business websites were much less likely to post notices than would be expected.
+Whilst this could be because construction companies were less likely to have been impacted by Covid from May, the more likely explanation is that construction company websites are not where clients of these companies go to get information - updates are likely to come through more channels.
+One surprising feature of this figure is that Wholesale and Retail Trade business websites were more likely to post notices than expected at the level of the UK; however Wholesale and Retail Trade business websites in Scotland were less likely.
```
This finding raises the question of "why". Perhaps say: this could be explained by differences in the products and services being sold by Scottish companies, a hypothesis that could be explored further through an analysis of their business descriptions and the text of their notices in future research.

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -139,3 +139,125 @@ We note that the estimates above are experimental and developed for illustrative
+
+In this section we present the results of performing topic modelling on the notices of Scottish businesses.
+
+[@tbl:topsbm_hierarchy] shows the topic hierarchy obtained. Topic level 3 was chosen as the initial level of analysis as the four topics yielded at level 4 are too broad and the 147 topics of level 2 are too fine.
```
... "with lower levels encompassing a larger number of more detailed topics".

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -139,3 +139,125 @@ We note that the estimates above are experimental and developed for illustrative
+</div>
+
+Figure [@fig:topic_trend_by_section] shows the topic activity in SIC sections over time, with the bottom of the plot showing the trends in the top 5 topics for each section.
+We see that four topics (0, 3, 6, 8) dominate the activity of most sections.
```
I would add a table with the topic keywords to illustrate the low level of informativeness.

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -139,3 +139,125 @@ We note that the estimates above are experimental and developed for illustrative
+> 12 May 2020 Civic Amenity Booking System by dgfarmer | posted in: Uncategorised | 0 Modus helps Civic Amenity Sites re-open after lockdown. We have been working with various Civic Amenity Site operators to produce a booking system suitable for use to regulate traffic at sites as they re-open after l...
+
+
+A further factor likely contributing to the poor outcomes from the topic modelling is the distribution of notice lengths - a large number of notices are extremely short and a large number are extremely long! Figure [@fig:notice_length] shows the notice length distribution - note the logarithmic x-scale.
```
remove exclamation mark

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -139,3 +139,125 @@ We note that the estimates above are experimental and developed for illustrative
+
+> The course is open. Please remember to adhere to the government social distancing 2 metre guidelines.
+
+Whereas the next notice is extremely long (and is also several months out of date),
```
Perhaps cut the text of the notice after a couple of paragraphs and say (the notice continues for another x words). Perhaps also worth removing any company identifiers ie "Ice Factor"?

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -262,6 +255,9 @@ Such notices are hard to algorithmically disambiguate from general statements ab
 
 ### Twitter analysis
 
+Having successfully scraped $34000$ of $45000$ Scottish websites we identified $12000$ websites with at least one twitter account candidate.
```
add: "in the glass data"

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -262,6 +255,9 @@ Such notices are hard to algorithmically disambiguate from general statements ab
 ### Twitter analysis
 
+Having successfully scraped $34000$ of $45000$ Scottish websites we identified $12000$ websites with at least one twitter account candidate.
+After filtering and matching we found twitter accounts for $9436$ Scottish business websites.
```
filtering and matching what?

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -303,25 +299,25 @@ Right: New twitter users each quarter (since 2019 Q1) by SIC section
- we do see a spike in the number of new users in April, May, and June 2020.
-However, the evidence for this is extremely weak as we are talking about 10-15 new users!
+ there is a spike in the number of new users in April, May, and June 2020.
+However, the evidence for this is extremely weak as the magnitude of the speak is about 10-15 new users!
```
Remove exclamation mark? Are we talking about 10-15 new users for each of those months or in total? (not that it makes much of a difference)

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -333,11 +329,11 @@ There is a hint of a small spike in more users stopping tweeting in March and Ap
-[@fig:tweets_open_close_norm] shows a large sharp spike in tweets containing "close" in the middle of March 2020 when the country first went into lockdown.
-Half as many tweets mentioned "close" in the week of April 15; however this way due to an increase in tweets mentioning "close" in Easter 2019, rather than an effect from April 2020.
+[@fig:tweets_open_close_norm] shows a large, sharp spike in tweets containing "close" in the middle of March 2020 when the country first went into lockdown.
+Half as many tweets mentioned "close" in the week of April 15; however this was due to an increase in tweets mentioning "close" in Easter 2019, rather than an effect from April 2020.
```
So does this mean that the drop in normalised mentions of close is partly caused by the spike in the denominator? Explain a bit further as this isn't clear

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -333,11 +329,11 @@ There is a hint of a small spike in more users stopping tweeting in March and Ap
+From October 2020, the number of tweets mentioning "close" rose again as more restrictions were imposed.
+There is an anomalous peak in tweets mentioning "close" in mid-February that would merit further investigation in future.
 
 <div class=altair s3_path="tweet_section_stack_terms.json" static_path="tweet_section_stack_terms.png" id="fig:tweet_section_stack_terms">Proportion of a weeks tweets (across all SIC sections) mentioning various terms, plotted by SIC section
```
This chart overlaps with the page title - perhaps decrease the height?

Juan-Mateos commented on reports/technical_report/4_results.md:
```diff
@@ -351,7 +347,7 @@ The sectoral heterogeneity is immediately apparent:
+ we do not see the context within which a term is used.
 For example, Section E begins tweeting about "supply" from August 2020; however this section relates to Water Supply and is not likely to be talking about supply issues for a product/service.
 
 
```
in the figure caption replace weeks with week's

Juan-Mateos commented on reports/technical_report/5_conclusions.md:
by restrictions, do you mean lockdown? Worth highlighting that the data looks useful for analysing Brexit impacts as well.

Juan-Mateos commented on reports/technical_report/5_conclusions.md:
```diff
@@ -60,16 +60,18 @@ Finally, we would like to incorporate additional secondary data into our analysi
 ### Micro analysis
 
 There are two principle significant investments required to improve the data quality of Covid-19 notices.
```
I would begin by saying: "Starting with the Covid-19 notices, there are..."

Juan-Mateos commented on reports/technical_report/5_conclusions.md:
```diff
@@ -60,16 +60,18 @@ Finally, we would like to incorporate additional secondary data into our analysi
 Secondly, better processing of the notices must be performed such that irrelevant notices are filtered out and relevant notices are classified into several categories such as "Business as usual", "Temporary closure", "Permanent closure", "Adaptation", "Partial closure" etc.
-This would likely require hand-annotation of the relevant sections of a subset of notices; training a model on these labels; extrapolating to the rest of the notices using the model; and performing a validation study on the results.
+This would require hand-annotation of a subset of notices; training a model on these labels; extrapolating to the rest of the notices using the model; and performing a validation study on the results.
 [Recent analysis](https://datasciencecampus.ons.gov.uk/extracting-text-data-from-business-website-covid-19-notices/)
```
Maybe also add a reference? Also note that the hyperlinks in the text are currently not coloured so hard to distinguish from plain text.

Juan-Mateos commented on reports/technical_report/5_conclusions.md:
```diff
@@ -60,16 +60,18 @@ Finally, we would like to incorporate additional secondary data into our analysi
 [Recent analysis](https://datasciencecampus.ons.gov.uk/extracting-text-data-from-business-website-covid-19-notices/)
  by the ONS has already piloted this hand labelling approach.
-While they find some promise in this approach; they are cautious about the biases of this approach.
+While they find some promise in this approach, they are cautious about the biases of this approach.
```
reduce repetition: "while they find some promise... they are cautious about potential biases"

Juan-Mateos commented on reports/technical_report/5_conclusions.md:
```diff
@@ -60,16 +60,18 @@ Finally, we would like to incorporate additional secondary data into our analysi
+There is probably more value in investing effort into social-media data over notice data given:
+ the myriad of issues with the Covid notice data,
+ the fact that the ONS is also exploring similar lines of enquiry,
+ and the ONS has a competitive edge with its access to microdata.
```
clarify what you mean: that they are able to match the covid notices to microdata in order to get better measures of Covid-19 impacts?

Juan-Mateos commented on reports/technical_report/5_conclusions.md:
```diff
@@ -60,16 +60,18 @@ Finally, we would like to incorporate additional secondary data into our analysi
+ the myriad of issues with the Covid notice data,
+ the fact that the ONS is also exploring similar lines of enquiry,
+ and the ONS has a competitive edge with its access to microdata.
 A potential scale-up option for the twitter data would be to put the twitter dataset into Elasticsearch to generate a search engine for the tweets of Scottish businesses which could be queried by policymakers to obtain lists of tweets relating to the query and interactive visualisations communicating the trends across 
```
I would reorder this paragraph and begin talking about scaling up in data collection and leave the searchable / explorable output for next section where we talk about visualisation options. Other potential research avenues I would consider:
* Analyse sectoral and geographical coverage
* Collect data for all the UK in order to compare the situation in Scotland: are there specific sectors / locations particularly vocal about issues with Covid-19?
* More sophisticated analysis of text in tweets

Juan-Mateos reviewed (commented):
A few minor comments and points for clarification.

I also have some suggestions about additional analysis and structure:

* I would move the twitter representation analysis (eg figures 30 and 31 and related text) to the methodology section for consistency with the Glass analysis. 
* I would break up the twitter analysis section into subsections eg: 
  * Evolution of activity
  * Entry and exit from Twitter
  * Evolution of activity by keyword
* I have also suggested moving next steps related to visualisation to the visualisation section and pointed out that hyperlinks are currently not highlighted in the PDF, which makes them hard to see.

Commit:
Executive summary

Commit:
Add URLS to interactives

Commit:
Edits in conclusions

bishax committed:
Address review comments

TODO:
- Render hyperlinks properly in PDF
- Update figures to use div syntax
- Use figure/table/section instead of fig/tbl/sec

bishax committed:
Merge branch '80_reporting' of github.com:nestauk/sg_covid_impact into 80_reporting

bishax committed:
Format/ref fixes;url color; include exec summary.

bishax committed:
Fix figurepaths to work locally and on s3. s3 report publish script.

bishax committed:
Exec summary edits.

bishax committed:
Exec summary brexit sentence.

bishax committed:
Font family change.

bishax committed:
Remove redundant header.

bishax committed:
Fix repro issues; update figures & export to S3;add altair-div to report

bishax committed:
Fix report generation to use metadata correctly.

Commit:
Report changes

Commit:
Fixed conflicts
